### ISSUE TITLE
Feature - Managed make parser

### ISSUE THREAD

**Issue by naveensrinivasan** [ossf/scorecard#1194]
**Is your feature request related to a problem? Please describe.**
The makefile parser is homegrown and harder to maintain 
**Describe the solution you'd like**
Explore in using https://github.com/mrtazz/checkmake

**Comment by chrismcgehee** [comment]
We do parsing of makefiles in scorecard? Or are you saying that [Makefile](https://github.com/ossf/scorecard/blob/main/Makefile) is hard to maintain?

**Comment by azeemshaikh38** [comment]
The feature ask is to have a Makefile parser similar to the GitHub workflow parser we currently have. It can potentially be used to analyze build scripts in repos.

**Comment by chrismcgehee** [comment]
Ok, it'd be used for things like https://github.com/ossf/scorecard/issues/884?

**Comment by azeemshaikh38** [comment]
Yes, #884 would be a great usecase for this. @laurentsimon fyi.

**Comment by laurentsimon** [comment]
Yes I want this parser for Makefiles! Let's make it for v5 release if nobody objects! Thanks for the find @naveensrinivasan 
@chrismcgehee if you want to do more work on pinning dependencies, feel free to assign to yourself :-)
